src/Core/Domain/ImageSettings/CommandHandler/DeleteImageTypeHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\ImageSettings\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsCommandHandler;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\DeleteImageTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Exception\ImageTypeNotFoundException;
+use PrestaShopBundle\Entity\Repository\ImageTypeRepository;
+
+/**
+ * Handles command that delete image type
+ */
+#[AsCommandHandler]
+final class DeleteImageTypeHandler implements DeleteImageTypeHandlerInterface
+{
+    public function __construct(
+        private readonly ImageTypeRepository $imageTypeRepository
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(DeleteImageTypeCommand $command): void
+    {
+        $imageType = $this->imageTypeRepository->find($command->getImageTypeId()->getValue());
+
+        if (null === $imageType->getId()) {
+            throw new ImageTypeNotFoundException(sprintf('Unable to find image type with id "%d" for deletion', $command->getImageTypeId()->getValue()));
+        }
+
+        $this->imageTypeRepository->delete($imageType);
```

Comment:
Same question as for the bulk, DB deletion is fine but what about the images themselves?